romeo/asset-contract/tests/asset_Commands.ts

```diff
@@ -0,0 +1,207 @@
+import { MintCommand } from "./asset_MintCommand.ts";
+import { TransferCommand } from "./asset_TransferCommand.ts";
+import { TransferCommand_NonOwner } from "./asset_TransferCommand_NonOwner.ts";
+
+export function AssetCommands(accounts: Map<string, Account>) {
+  const data = getBitcoinTxData();
+  const cmds = [
+    // BurnCommand
+    fc
+      .record({
+        sender: fc.constant(accounts.get("deployer")!),
+        amount: fc.integer({ min: 1 }),
+        wallet: fc.constantFrom(...accounts.values()).filter((a: Account) => a.address !== accounts.get("deployer")!.address),
+        params: fc.constantFrom(...data),
+      })
+      .map((
+        r: {
+          sender: Account;
+          amount: number;
+          wallet: Account;
+          params: BitcoinTxData;
+        },
+      ) =>
+        new BurnCommand(
+          r.sender,
+          r.amount,
+          r.wallet,
+          r.params,
+        )
+      ),
+
+    // GetBalanceCommand
+    fc
+      .record({
+        sender: fc.constantFrom(...accounts.values()),
+        wallet: fc.constantFrom(...accounts.values()),
+      })
+      .map((
+        r: {
+          sender: Account;
+          wallet: Account;
+        },
+      ) =>
+        new GetBalanceCommand(
+          r.sender,
+          r.wallet,
+        )
+      ),
+
+    // GetTotalSupplyCommand
+    fc
+      .record({
+        sender: fc.constantFrom(...accounts.values()),
+      })
+      .map((
+        r: {
+          sender: Account;
+        },
+      ) =>
+        new GetTotalSupplyCommand(
+          r.sender,
+        )
+      ),
+
+    // MintCommand
+    fc
+      .record({
+        sender: fc.constant(accounts.get("deployer")!),
+        amount: fc.integer({ min: 1 }),
+        wallet: fc.constantFrom(...accounts.values()).filter((a: Account) => a.address !== accounts.get("deployer")!.address),
+        params: fc.constantFrom(...data),
+      })
+      .map((
+        r: {
+          sender: Account;
+          amount: number;
+          wallet: Account;
+          params: BitcoinTxData;
+        },
+      ) =>
+        new MintCommand(
+          r.sender,
+          r.amount,
+          r.wallet,
+          r.params,
+        )
+      ),
+
+    // TransferCommand
+    fc
+      .record({
+        sender: fc.constantFrom(...accounts.values()),
+        amount: fc.integer({ min: 1 }),
+        wallet: fc.constantFrom(...accounts.values()),
+      })
+      .map((
+        r: {
+          sender: Account;
+          amount: number;
+          wallet: Account;
+        },
+      ) =>
+        new TransferCommand(
+          r.sender,
+          r.amount,
+          r.wallet,
+        )
+      ),
+
+    fc
+      .record({
+        sender: fc.constantFrom(...accounts.values()),
+        amount: fc.integer({ min: 1 }),
+        holder: fc.constantFrom(...accounts.values()),
+        wallet: fc.constantFrom(...accounts.values()),
```

Comment:
Just curiosity:

Are there problems if the accounts sending and holding are the same? If we needed them to be unique but all from the list how would we do that?

Reply:
It be checked before running the test: https://github.com/stacks-network/sbtc/pull/152/files#diff-37f7b3538c8e0981f5d429a974ce04c59b31c740ae25826d8da7cf475f39b29aR28

Reply:
So then we'd still be setting up the test incorrectly but then nixing it before it runs?